### DISCUSSION TITLE
[SPA] Popover position for a document with dynamic scrolling

### DISCUSSION THREAD

**Discussion by DmitryShcheklein** [floating-ui/floating-ui#2370] [Q&A]
Hello, i have SPA application. I added a style fix for page so that is doesn't shift:
```
html {
  margin-left: calc(100vw - 100%);
}
```
But my Popover position calculated wrong. How can i fix it?
My code:
https://codesandbox.io/s/beautiful-currying-m86zd6?file=/src/Popover.tsx:1128-1164

![2023-06-10_17-25-43](https://github.com/floating-ui/floating-ui/assets/16781081/4c785dca-e35d-4456-a81d-30cc80a9fc20)
![2023-06-10_17-25-29](https://github.com/floating-ui/floating-ui/assets/16781081/3feeec2a-b30e-41fe-bf98-97d82624e681)

**Comment by atomiks** [comment]
This just seems like a bug with the `dom` library not taking margin on `html` into account. Can you reproduce it with `@floating-ui/dom` and make this into an issue if so?

A workaround is to use a middleware to add `x + scrollbarWidth` to the coordinates for this layout, for now. Or you can put `overflow-y: scroll` on `<html>` and avoid `margin`, which should achieve the same thing.

**Comment by DmitryShcheklein** [comment]
1. Hmm, ok. I'll try to repeat this a bit later with @floating-ui/dom

2. I can't avoid margin :(
Use middleware is good, but I did not manage to achieve the desired result with popup and arrow. Am I in the right direction?)
https://codesandbox.io/s/beautiful-currying-m86zd6?file=/src/Popover.tsx:1279-1310

**Reply by atomiks** [reply]
2. Why not? How is it that there are many SPAs that haven't had this positioning problem since they haven't had to resort to margin? Why doesn't `overflow-y: scroll` work?

**Reply by atomiks** [reply]
I think the reason things end up wrong is `.getBoundingClientRect()` will return incorrect `x` values for all elements with the margin on `html` (needs to be considered), so I don't really think it's a good idea to use to begin with since all `.getBoundingClientRect()` usage with absolute elements ends up breaking. Especially when you consider RTL scrollbars on the window, so you need to invert the value, etc. Becomes a mess quickly.

**Reply by DmitryShcheklein** [reply]
> 2. Why not? How is it that there are many SPAs that haven't had this positioning problem since they haven't had to resort to margin? Why doesn't `overflow-y: scroll` work?

"Overflow-y: scroll" it is a good simple decision, but I have a problem with open modal when use `Overflow-y: scroll` on html. My page is jumping (because changed it on overflow:hidden) and my designer will be unhappy 🥲
I'll try again..
Thank you very much!

**Reply by atomiks** [reply]
If you use Floating UI for the Dialog too, then you can use `FloatingOverlay` `lockScroll` prop which handles that for you (adds padding for the scrollbar): https://floating-ui.com/docs/FloatingOverlay#lockscroll

**Reply by DmitryShcheklein** [reply]
Oh, I didn't use Floating UI for the Dialog, didn't know, thanks. I'll try. Maybe this will solve the jump problem.

**Reply by DmitryShcheklein** [reply]
@atomiks, I have "page-top-jump" effect if using FloatingOverlay or my Dialog body fix, because they change style `Overflow-y: scroll` to `Overflow: hidden`
